code/chapters/09-mapping.py

```diff
@@ -74,6 +74,77 @@
 # - Panel: allows you to create applications/dashboards
 #
 # ### GeoPandas explore
+import io
+import pandas as pd
+import requests
+
+pd.set_option('display.max_columns', None)
+
+CRS = 'EPSG:32630'
+
+def get_databuffer(uri, encoding='utf-8'):
+    """Download data from URI and returns as an StringIO buffer"""
+    r = requests.get(uri, timeout=10)
+    return io.StringIO(str(r.content, encoding))
+
+def download_data(uri, filepath='./NaPTAN.csv'):
+    """Download data from URI and returns save to file system """
+    with open(filepath, 'wb') as fout:
+        data = requests.get(uri)
+        fout.write(data.content)
+    return True
+
+# NaPTAN data service
+URI='https://multiple-la-generator-dot-dft-add-naptan-prod.ew.r.appspot.com/v1/access-nodes?dataFormat=csv'
+
+BUFFER = get_databuffer(URI)
+DF1 = pd.read_csv(BUFFER, low_memory=False)
+DATA = DF1[['Longitude', 'Latitude']].values
+POINTS = gpd.GeoSeries.from_xy(*DATA.T, crs='WGS84')
+NaPTAN = gpd.GeoDataFrame(data=DF1, geometry=POINTS)
+NaPTAN = NaPTAN.dropna(how='all', axis=1)
```

Comment:
Makes sense, need to drop the NAs.

Reply:
This is a fairly strict criteria for the NA drop, a how='any' tends to drop useful stuff